### DISCUSSION TITLE
Setting to see InnerException?

### DISCUSSION THREAD

**Discussion by JSatterly** [oqtane/oqtane.framework#2434] [Q&A] [answered]
When errors are logged, is there a setting that will also push the inner exception or more detail into the Log.Exception field? I was provoking an Error by adding a user with a non unique email and the popup display page only mentioned a username error but no additional details were saved. I tried different appsettings for the loggers including Debug and Information but the field remained empty in the Log table.  Looking at the ILogManager, only a couple of the overrides sends the exception?
Thanks.

**Comment by sbwalker** [comment, marked as answer]
In cases where there is an unhandled exception the framework has methods for passing the Exception:

client-side (.razor):

public async Task LogError(Exception exception, string message, params object[] args);

server-side (.cs)

void Log(LogLevel level, object @class, LogFunction function, Exception exception, string message, params object[] args);

Both of these methods are handled by LogManager which uses Exception.ToString() internally to save the details of the exception (which includes the InnerException) to the database.

**Reply by JSatterly** [reply]
OK thanks, I expected to see it each time. I'll watch for it when needed.

**Reply by sbwalker** [reply]
Your comment about the add user API behavior is still relevant though... Oqtane uses .NET Identity and rather than throw exceptions, the Identity methods trap them internally and return a custom object that indicates success or failure and also includes any validation errors in a custom collection. Oqtane is currently using the Boolean but ignoring the custom collection... but it probably should also be sending the detailed info to the event log to provide better insights.

**Reply by JSatterly** [reply]
That would help, since the feedback was that the pw was wrong or the username not unique when the reality was that the email wasn't unique. I know login fails aren't supposed to give back too much info but behind the scenes or when registering would be helpful.
